src/webgpu/util/math.ts

```diff
@@ -559,3 +559,43 @@ export function hexToF64(h32: number, l32: number): number {
   const f64Arr = new Float64Array(u32Arr.buffer);
   return f64Arr[0];
 }
+
```

Comment:
For n-th time, I feel like having a proper cartesian product operation is needed. This time I think it is actually needed and not just me over-designing things.

There is sexy way to do this using map and reduce, but honestly I couldn't reason through how it all holds together in TS, so I did it more explicitly, but still functionally. I may in the future come back and rewrite this using more language specific facilities if I can figure out how to do it without it be completely opaque/black magic-y.